From fc38b73e2556211d698849b78a4623dc163b7d49 Mon Sep 17 00:00:00 2001
From: Arnaud Ferraris <arnaud.ferraris@gmail.com>
Date: Wed, 8 Dec 2021 23:43:08 +0100
Subject: [PATCH 3/4] arm64: dts: rk3399-pinephone-pro: add modem RI pin

Taht way the modem can wake the phone on incoming calls/messages.

Signed-off-by: Arnaud Ferraris <arnaud.ferraris@gmail.com>
---
 .../dts/rockchip/rk3399-pinephone-pro.dts     | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index c9365e604..b3b091ea7 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -160,6 +160,21 @@ power {
 		};
 	};
 
+	gpio-key-ri {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&ri_pin>;
+
+		ring_indicator: ring-indicator {
+			label = "ring-indicator";
+			linux,can-disable;
+			linux,code = <KEY_WAKEUP>;
+			gpios = <&gpio0 RK_PA1 GPIO_ACTIVE_LOW>;
+			wakeup-event-action = <EV_ACT_ASSERTED>;
+			wakeup-source;
+		};
+	};
+
 	// in1  - digital mic daughhterboard
 	// in2  - headset mic
 	// in3  - modem output  (muxed with mono)
@@ -1150,6 +1165,10 @@ flash_pins: flash-pins {
 	};
 
 	modem {
+		ri_pin: ri-pin {
+			rockchip,pins = <0 RK_PA1 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+
 		vcc_4g_5v_en: vcc-4g-5v-en-pin {
 			rockchip,pins = <1 RK_PC7 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
-- 
2.34.1

